pkg/cloudprovider/README.md

```diff
@@ -1,30 +0,0 @@
-### Deprecation Notice
```

Comment:
This README may still be useful in the legacy-cloud-provider repo. May want to re-add it in a follow-up PR

Reply:
I will add it back in a follow up PR. Thanks

Reply:
Add back README at [92827](https://github.com/kubernetes/kubernetes/pull/92872)